src/mongo/test-functional/MongoDbConnection_GetCollection_Tests.cs

```diff
@@ -16,19 +18,59 @@ public MongoDbConnection_GetCollection_Tests()
         [Fact]
         public void GetCollection_usesGivenCollectionName()
         {
-            var actual = connection.GetCollection<Document>("kewl_stuff");
+            var actual = connection.GetCollection<DocumentWithCollectionSpecified>("kewl_stuff");
 
-            Assert.IsAssignableFrom<IMongoCollection<Document>>(actual);
+            Assert.IsAssignableFrom<IMongoCollection<DocumentWithCollectionSpecified>>(actual);
             Assert.Equal("kewl_stuff", actual.CollectionNamespace.CollectionName);
         }
 
         [Fact]
-        public void GetCollection_usesNameOfDocumentType_ifNotGivenACollectionName()
+        public void GetCollection_usesCollectionNameDefinedInEntityAttribute_ifParamIsNull()
```

Comment:
**praise**: nice to see that you also updated the name of the test :)

That is something that many tend to forget :)